### DISCUSSION TITLE
Custom Search Help

### DISCUSSION THREAD

**Discussion by JakeHenshall** [statamic/cms#4915] [Q&A]
Hi. Im pretty new to Statamic but really enjoying my time with it soo far but i'm a little stuck on how best to achieve this.

Im using a solution called supercontrol and this is how they say to get search results:

As a result, when populating your website search results and property pages the information can come directly from your own database. You only need to query the API for the property list and prices. For example: https://api.supercontrol.co.uk/xml/filter3.asp?siteID=46111&startdate=2021-12-21&numbernights=7&basic_details=1

I currently have the properties as a collection using the fantastic YumYum plugin with help from Duncan.

What i'm trying todo is something very akin to this site -> https://luxurycoastal.co.uk/properties?region=10131&startdate=2022-05-13&numbernights=7&datevar=3&sleeps=2 which also uses supercontrol.

Is there a way to connect the results that are shown from this link https://api.supercontrol.co.uk/xml/filter3.asp?siteID=46111&startdate=2021-12-21&numbernights=7&basic_details=1 to display the content from the collection based on different searches.

Apologies if something similar has already been asked.

Any help is greatly appreciated.

Please let me know if I need to describe anything in more detail.

Regards, Jake.

**Comment by edalzell** [comment]
Sorry I don't understand. You HAVE the data in Statamic, so what do you need the external bit for?

**Comment by JakeHenshall** [comment]
@edalzell Thanks for getting back.

I believe that supercontrol requires you to directly query their API for the available properties and prices information that is returned in XML Format.

I only have the data for properties, not if they are available or prices.

The data I already have in my statamic site it from this url -> https://api.supercontrol.co.uk/xml/filter3.asp?siteID=46111

To show the properties that are available it needs to use this link -> https://api.supercontrol.co.uk/xml/filter3.asp?siteID=46111&startdate=2021-12-21&numbernights=7&basic_details=1

I'm basically wondering what the 'correct' way in statamic is to:

Read the custom URL created by the form (It will have filters) e.g (https://api.supercontrol.co.uk/xml/filter3.asp?siteID=46111&startdate=2021-12-21&numbernights=7&basic_details=1) to parse the results from the above url onto a results page.

I hope that clears it up a little more, apologies if not :{.

Regards,

Jake.

**Reply by jelleroorda** [reply]
You can [create a custom tag](https://statamic.dev/extending/tags) which would make a request to that URL and shows the results.

**Reply by JakeHenshall** [reply]
> 

Hey man thanks, I will look into the custom tag option, thanks.

**Comment by edalzell** [comment]
Ya so you might need to do what I did w/ shopify and make my own Algolia indexer. You could also use some custom search transformers to get the search stuff in

**Reply by JakeHenshall** [reply]
@edalzell Thanks for getting back and mentioning those 2 ways, I will look into the custom tag option first and then see how thats goes. Thanks.

**Reply by JakeHenshall** [reply]
@edalzell The tag seems to be doing what I'd like, i'm just having trouble positioning the output on the search page. The Tag content is being pushed above everything else, The content is being looped in a for loop and being echoed out. Is there anything that I can do to sort this, or do you need me to create another post? Thanks Jake.

**Reply by edalzell** [reply]
> @edalzell The tag seems to be doing what I'd like, i'm just having trouble positioning the output on the search page. The Tag content is being pushed above everything else, The content is being looped in a for loop and being echoed out. Is there anything that I can do to sort this, or do you need me to create another post? Thanks Jake.

I don't understand, can you show us what you mean?

**Reply by JakeHenshall** [reply]
@edalzell Sure, 

![Props](https://user-images.githubusercontent.com/4720782/147619282-5eb3d70f-f84d-4be1-9357-4f80e2666076.png)

![Inspect](https://user-images.githubusercontent.com/4720782/147619292-4db98514-8ddf-4a1c-aea0-b4b1fb0cc105.png)

If I just return some text it displays in the correct place, which is under the form where i'm able to style it with Tailwind.

Hope that makes more sense.

Thanks, Jake.

**Reply by edalzell** [reply]
you're in complete charge of the front end, put the results somewhere else.

FYI, you don't need to tag me.
